### PR TITLE
feat: Basic implementation of reprocessing, cloning events

### PR DESCRIPTION
**Pull request by untitaker**
If a user fails to provide debug symbols, their native stacktraces are worthless. Sometimes they realize this too late when the event has already been saved.

Ideally they would be able to just "refresh" an event, which would go through stacktrace symbolication again and be updated in place, with the event moving between groups if the stacktrace is significantly different. Unfortunately Snuba does not support that, so we resorted to a more barebones implementation: When the user hits the reprocess button, the event is duplicated, assigned a new event ID and timestamp, and that copy is inserted into the evnet processing pipeline again. That's basically what this PR does:

* Additional code in preprocess_event to backup the event payload before processing (but after data scrubbing)
* Reprocessing endpoints that trigger celery task to reprocess entire issue or a single event
* In either case the payload backup is loaded, the attachments are loaded from filestore into the attachments_cache, and the event goes through the same exact pipeline again.
* ...while being on the reprocessing queue such that backlogs don't interfere with "live" event ingestion (note: that queue already exists for the "old reprocessing" feature that basically prevents the event from saving if the stacktrace is bad, and gives the user a button to "resume the pipeline/restart processing")

Open TODOs that block merging:

- [x] Make this play nice with projects that have the old reprocessing enabled
- [x] consider creating new queues for this new reprocessing (reprocessing2) -- **not for now**
- [x] fix symbolicator caching issues in testsuite

### CODE REVIEW

File: src/sentry/tasks/reprocessing2.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@instrumented_task(
+    name="sentry.tasks.reprocessing2.reprocess_event",
+    queue="events.reprocessing.preprocess_event",  # XXX: dedicated queue
+    time_limit=30,
+    soft_time_limit=20,
+)
+def reprocess_event(project_id, event_id, start_time):
+    from sentry.reprocessing2 import reprocess_event
+
+    reprocess_event(project_id=project_id, event_id=event_id, start_time=start_time)
```
Comment (tonyo) [inline src/sentry/tasks/reprocessing2.py:59]: Let's.. maybe rename this function?
Looks like as recursion at first.

File: tests/sentry/tasks/test_store.py [hunk 2]
```diff
@@ -104,7 +92,7 @@ def mock_metrics_timing():
 def test_move_to_process_event(
     default_project, mock_process_event, mock_save_event, mock_symbolicate_event, register_plugin
 ):
-    register_plugin(BasicPreprocessorPlugin)
+    register_plugin(globals(), BasicPreprocessorPlugin)
```
Comment (tonyo) [inline tests/sentry/tasks/test_store.py:95]: Not sure what's happening here

File: src/sentry/eventstore/processing/base.py [hunk 3]
```diff
@@ -21,19 +25,23 @@ def __init__(self, inner, timeout=DEFAULT_TIMEOUT):
...
-    def get(self, key):
+    def get(self, key, unprocessed=False):
+        if unprocessed:
+            key = _get_unprocessed_key(key)
         return self.inner.get(key)
 
     def delete_by_key(self, key):
-        return self.inner.delete(key)
+        self.inner.delete(key)
+        self.inner.delete("{}:u".format(key))
```
Comment (tonyo) [inline src/sentry/eventstore/processing/base.py:42]: 1. We can use `_get_unprocessed_key` here
2. This basically means another unconditional call to Redis, right? Any way to restrict it only to specific cases, somehow?

**Review by tonyo — COMMENTED** [review]
No major objections so far; I think we can safely reuse the existing reprocessing queues since they are used pretty rarely.

File: tests/sentry/tasks/test_store.py (hunk 2 above)
Comment (untitaker) [reply tests/sentry/tasks/test_store.py:95]: This is just wonky test setup to register and unregister plugins. `register_plugin` used to be defined in this module (and just called `globals()` itself), but somehow I couldn't get it to work in conftest.py.

The reason we use globals is because the plugin manager needs a valid import-path to import the module from, so we temporarily provide that in tests.

File: src/sentry/eventstore/processing/base.py (hunk 3 above)
Comment (untitaker) [reply src/sentry/eventstore/processing/base.py:42]: If we change this to `delete_many` would it be ok to run it unconditionally?

File: src/sentry/tasks/reprocessing2.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/tasks/reprocessing2.py:59]: is this better?

File: src/sentry/eventstore/processing/base.py (hunk 3 above)
Comment (tonyo) [reply src/sentry/eventstore/processing/base.py:42]: I don't think we can use `delete_many`, because it's not part of the interface now?

File: src/sentry/eventstore/snuba/backend.py [hunk 4]
```diff
@@ -194,8 +194,6 @@ def get_event_by_id(self, project_id, event_id):
         if event.get_event_type() != "transaction":
             result = snuba.raw_query(
                 selected_columns=["group_id"],
-                start=event.datetime,
-                end=event.datetime + timedelta(seconds=1),
```
Comment (tonyo) [inline src/sentry/eventstore/snuba/backend.py:198]: Are we sure we don't need this anymore?

File: src/sentry/tasks/reprocessing2.py (hunk 1 above)
Comment (tonyo) [reply src/sentry/tasks/reprocessing2.py]: Looks less confusing to me, thanks 👍

File: src/sentry/eventstore/processing/base.py (hunk 3 above)
Comment (untitaker) [reply src/sentry/eventstore/processing/base.py:42]: ah yes inner is not a django cache. I can extend it?

File: src/sentry/eventstore/snuba/backend.py (hunk 4 above)
Comment (untitaker) [reply src/sentry/eventstore/snuba/backend.py:198]: wow thanks, fixed

File: src/sentry/eventstore/processing/base.py (hunk 3 above)
Comment (untitaker) [reply src/sentry/eventstore/processing/base.py:42]: it's tricky to do something faster than delete_many, as checking for the feature in here is equally expensive than just attempting to delete

File: src/sentry/api/endpoints/group_reprocessing.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+def reprocess_group_scenario(runner):
+    group = Group.objects.filter(project=runner.default_project).first()
+    runner.request(method="PUT", path="/issues/%s/reprocessing/" % group.id)
+
+
+class GroupReprocessingEndpoint(GroupEndpoint):
+    doc_section = DocSection.EVENTS
+
+    @attach_scenarios([reprocess_group_scenario])
+    def put(self, request, group):
```
Comment (jan-auer) [inline src/sentry/api/endpoints/group_reprocessing.py:22]: Should this be `POST` rather? At the moment, reprocessing is not idempotent (it does create new events every time), so this resembles `POST` semantics.

File: src/sentry/api/urls.py [hunk 6]
```diff
@@ -341,6 +343,7 @@
         GroupNotesDetailsEndpoint.as_view(),
     ),
     url(r"^(?P<issue_id>[^\/]+)/hashes/$", GroupHashesEndpoint.as_view()),
+    url(r"^(?P<issue_id>[^\/]+)/reprocessing/$", GroupReprocessingEndpoint.as_view()),
```
Comment (jan-auer) [inline src/sentry/api/urls.py:346]: It seems that we currently have such actions on index endpoints (e.g. `/issues/<id>/?reprocess=true`). That said, since we want to query for reprocessing status, it makes sense to introduce a new endpoint.

File: src/sentry/conf/server.py [hunk 7]
```diff
@@ -1954,3 +1962,5 @@ def get_sentry_sdk_config():
 }
 
 SENTRY_USE_UWSGI = True
+
+SENTRY_REPROCESSING_ATTACHMENT_CHUNK_SIZE = 10 ** 7
```
Comment (jan-auer) [inline src/sentry/conf/server.py:1966]: Current chunk size in Redis is `2**10` (1MiB), not `10**7` (10MB). I was hoping to remove a special configuration for this, but since that is currently controlled only by Relay, we cannot reuse something existing.

File: src/sentry/eventstream/snuba.py [hunk 8]
```diff
@@ -147,9 +147,11 @@ def start_delete_groups(self, project_id, group_ids):
 
         return state
 
-    def end_delete_groups(self, state):
+    def end_delete_groups(self, state, cutoff_datetime=None):
```
Comment (jan-auer) [inline src/sentry/eventstream/snuba.py:150]: Is this parameter still being used? It seems that this is dead code and can be reverted.

File: src/sentry/models/eventattachment.py [hunk 9]
```diff
@@ -5,7 +5,12 @@
 from django.db import models
 from django.utils import timezone
 
-from sentry.db.models import BoundedBigIntegerField, FlexibleForeignKey, Model, sane_repr
+from sentry.db.models import (
+    BoundedBigIntegerField,
+    FlexibleForeignKey,
+    Model,
+    sane_repr,
+)
```
Comment (jan-auer) [inline src/sentry/models/eventattachment.py:13]: _nit_: Changes in this file seem unrelated could be reverted to minimize the diff.

File: src/sentry/reprocessing2.py [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+    ):
+        data = event_processing_store.get(
+            cache_key_for_event({"project": project.id, "event_id": event_id}), unprocessed=True
+        )
+        if data is None:
+            return
+
+    with sentry_sdk.start_span(op="sentry.reprocessing2.save_unprocessed_event.set_nodestore"):
+        node_id = _generate_unprocessed_event_node_id(project_id=project.id, event_id=event_id)
+        nodestore.set(node_id, data)
```
Comment (jan-auer) [inline src/sentry/reprocessing2.py:44]: _nit_: Current node store writes are located in EventManager code. Can we move this over to consolidate, and just leave the reprocessing logic here? If you prefer to keep it here, that's fine with me, too.

File: src/sentry/reprocessing2.py [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+    # Take unprocessed data from old event and save it as unprocessed data
+    # under a new event ID. The second step happens in pre-process. We could
+    # save the "original event ID" instead and get away with writing less to
+    # nodestore, but doing it this way makes the logic slightly simpler.
+
+    # Step 1: Fix up the event payload for reprocessing and put it in event
+    # cache/event_processing_store
+    tags = dict(data.get("tags") or ())
+    orig_event_id = tags["original_event_id"] = data["event_id"]
```
Comment (jan-auer) [inline src/sentry/reprocessing2.py:83]: Putting this in tags is fine, as it gives us a simple way to search by it and jump to the original event. However, we might want to consider adding a dedicated attribute in the future similar to "release" so that we can apply better special handling.

File: src/sentry/reprocessing2.py [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+
+    # XXX: Only reset received
+    data["timestamp"] = data["received"] = start_time
+
+    cache_key = event_processing_store.store(data)
+
+    # Step 2: Copy attachments into attachment cache
+    queryset = EventAttachment.objects.filter(
+        project_id=project_id, event_id=orig_event_id
+    ).select_related("file")
```
Comment (jan-auer) [inline src/sentry/reprocessing2.py:101]: In a follow-up: We can consider to only load attachments required for processing, that is, crash reports. For other attachments, we can do one of:

1. Copy file records in save_event without accessing blob store, if we keep duplicating the events.
2. Do nothing, if we choose to mutate events.

File: src/sentry/reprocessing2.py [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+                    id=attachment_id,
+                    name=attachment.name,
+                    # XXX: Not part of eventattachment model, but not strictly
+                    # necessary for processing
+                    content_type=None,
+                    type=attachment.file.type,
+                    chunks=chunk_index,
+                    size=size,
+                )
+            )
```
Comment (jan-auer) [inline src/sentry/reprocessing2.py:143]: _nit_: Move the file copy process into a utility function for readability.

File: src/sentry/static/sentry/app/components/events/eventEntries.jsx [hunk 14]
```diff
@@ -195,7 +195,12 @@ class EventEntries extends React.Component {
       <div className={className} data-test-id="event-entries">
         {!objectIsEmpty(event.errors) && (
           <ErrorContainer>
-            <EventErrors event={event} />
+            <EventErrors
+              event={event}
+              orgId={organization.slug}
+              project={project}
+              issueId={group.id}
```
Comment (jan-auer) [inline src/sentry/static/sentry/app/components/events/eventEntries.jsx:202]: This is something for UI folks like @priscilawebdev or @matejminar. I think we have context adapters (like `withProjectContext` or similar), so that you can access the ambient project and group without having to pass it through. Not sure if that's desired here, though.

File: src/sentry/reprocessing2.py [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+        cache_key=cache_key, start_time=start_time, event_id=event_id
+    )
+
+
+def is_reprocessed_event(data):
+    return bool(_get_original_event_id(data))
+
+
+def _get_original_event_id(data):
+    return dict(data.get("tags") or ()).get("original_event_id")
```
Comment (jan-auer) [inline src/sentry/reprocessing2.py:159]: _nit_: The event model has a `get_tag` function, which we might want to consider to factor out and reuse here. It seems that in most cases you have an event model already, so maybe you can just call `get_tag` directly?

File: src/sentry/tasks/reprocessing2.py [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+        )
+    )
+
+    if not events:
+        return
+
+    for event in events:
+        reprocess_event.delay(
+            project_id=project_id, event_id=event.event_id, start_time=start_time,
+        )
```
Comment (jan-auer) [inline src/sentry/tasks/reprocessing2.py:43]: It will be a challenge to detect when *all* events have finished reprocessing for this group. Let's please start evaluating which options we have there, as this will have strong influence on the design of this job.

**Review by jan-auer — APPROVED** [review]
LGTM, with a few non-blocking comments below. 

Would appreciate a quick review from @matejminar regarding UI changes, and someone from backend/tooling teams to double-check the changes in the test infra.

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 17]
```diff
@@ -0,0 +1,242 @@
...
+          </li>
+          <li>
+            {t(
+              'If you have provided missing symbols please wait at least 1 hour before attempting to re-process. This is a limitation we will try to get rid of.'
+            )}
+          </li>
+        </ul>
+      </Body>
+      <div className="modal-footer">
+        <Button onClick={this.onReprocessGroup}>
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/errors.tsx:151]: You can wrap this in a <ButtonBar gap={1}> so that you don't have to do inline marginLefts.

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 18]
```diff
@@ -0,0 +1,242 @@
...
+          {t('Reprocess all events in issue')}
+        </Button>
+        <Button style={{marginLeft: space(1)}} onClick={this.onReprocessEvent}>
+          {t('Reprocess single event')}
+        </Button>
+        <Button style={{marginLeft: space(1)}} onClick={closeModal}>
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/errors.tsx:157]: I don't see how does this look, but should these buttons be color-coded? E.g. one action be primary, etc..

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 19]
```diff
@@ -0,0 +1,242 @@
...
+          {t('Cancel')}
+        </Button>
+      </div>
+    </React.Fragment>
+  );
+
+  render() {
+    const {event, project} = this.props;
+    const features = new Set(project.features);
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/errors.tsx:166]: You don't have to create a set anymore. You can use `includes` directly on `features` array. So you would check for feature flag like this:
`project.features.includes('reprocessing-v2')`

File: src/sentry/api/endpoints/group_reprocessing.py (hunk 5 above)
Comment (untitaker) [reply src/sentry/api/endpoints/group_reprocessing.py:22]: I've seen PUT used to create new entities, so I thought "let's make a reprocessing task an entity". TIL PUT is supposed to be idempotent. Will fix.

File: config/symbolicator/config.yml [hunk 20]
```diff
@@ -0,0 +1,11 @@
+bind: "0.0.0.0:3021"
+logging:
+  level: "trace"
```
[the remark this answers is not in the document]
Comment (untitaker) [reply config/symbolicator/config.yml:3]: Sure

File: src/sentry/api/urls.py (hunk 6 above)
Comment (untitaker) [reply src/sentry/api/urls.py:346]: yeah that was the idea PUT/POST for creating a "task entity" and GET on the same endpoint (or rather a subpath) for polling

File: src/sentry/conf/server.py (hunk 7 above)
Comment (untitaker) [reply src/sentry/conf/server.py:1966]: I think I asked about this in Slack fwiw and got 10mb as answer. Will change to 1mb

File: src/sentry/eventstream/snuba.py (hunk 8 above)
Comment (untitaker) [reply src/sentry/eventstream/snuba.py:150]: damn, yeah will fix

File: src/sentry/models/eventattachment.py (hunk 9 above)
Comment (untitaker) [reply src/sentry/models/eventattachment.py:13]: thanks, yeah I used to have attachment type added here until I realized which table I am about to migrate (and until I realized it's on File so we're good for now)

File: src/sentry/reprocessing2.py (hunk 10 above)
Comment (untitaker) [reply src/sentry/reprocessing2.py:44]: I'll keep it here to keep `_generate_unprocessed_event_node_id` private

File: src/sentry/reprocessing2.py (hunk 11 above)
Comment (untitaker) [reply src/sentry/reprocessing2.py:83]: yeah this is an absolutely massive hack. this should really be its own snuba column but I was trying to avoid binding any time from S&S

File: src/sentry/reprocessing2.py (hunk 12 above)
Comment (untitaker) [reply src/sentry/reprocessing2.py:101]: if we just copy the file records won't blobstore have to have a reference count such that the blob is deleted on the second event deletion, not the first?

File: src/sentry/reprocessing2.py (hunk 13 above)
Comment (untitaker) [reply src/sentry/reprocessing2.py:143]: will do

File: src/sentry/reprocessing2.py (hunk 15 above)
Comment (untitaker) [reply src/sentry/reprocessing2.py:159]: in `create_failed_event` I have no event model unfortunately

File: src/sentry/tasks/reprocessing2.py (hunk 16 above)
Comment (untitaker) [reply src/sentry/tasks/reprocessing2.py:43]: Yeah. I think a redis set (set-the-datastructure, not set-the-command) full of pending event IDs might be good, and then we can poll from the UI until it is empty

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 21]
```diff
@@ -0,0 +1,242 @@
...
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    if (this.state.isOpen !== nextState.isOpen) {
+      return true;
+    }
+    return this.props.event.id !== nextProps.event.id;
+  }
+
+  toggle = () => {
+    this.setState({isOpen: !this.state.isOpen});
```
[the remark this answers is not in the document]
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:60]: will do

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 22]
```diff
@@ -0,0 +1,242 @@
...
+  onReprocessStart = () => {
+    openModal(this.renderReprocessModal);
+  };
+
+  renderReprocessModal = ({Body, closeModal}) => (
+    <React.Fragment>
+      <Body>
+        <p>
+          {t(
+            'You can choose to re-process events to see if ' +
```
[the remark this answers is not in the document]
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:122]: wanted to line-wrap, but will remove

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 23]
```diff
@@ -0,0 +1,242 @@
...
+  uniqueErrors = errors => uniqWith(errors, isEqual);
+
+  onReprocessEvent = async () => {
+    const {orgId, project, event} = this.props;
+    const endpoint = `/projects/${orgId}/${project.slug}/events/${event.id}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing event\u2026'));
+
+    try {
+      await this.props.api.requestPromise(endpoint, {
```
[the remark this answers is not in the document]
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:72]: will do

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 24]
```diff
@@ -0,0 +1,242 @@
...
+  };
+
+  onReprocessGroup = async () => {
+    const {orgId, issueId} = this.props;
+    const endpoint = `/organizations/${orgId}/issues/${issueId}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing issue\u2026'));
+
+    try {
+      await this.props.api.requestPromise(endpoint, {
```
[the remark this answers is not in the document]
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:96]: will do

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 25]
```diff
@@ -0,0 +1,242 @@
...
+    browserHistory.push(
+      `/organizations/${orgId}/issues/?query=tags[original_group_id]:${issueId}`
+    );
+  };
+
+  onReprocessStart = () => {
+    openModal(this.renderReprocessModal);
+  };
+
+  renderReprocessModal = ({Body, closeModal}) => (
```
[the remark this answers is not in the document]
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:117]: will do

File: src/sentry/static/sentry/app/components/events/errors.tsx (hunk 17 above)
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:151]: will do

File: src/sentry/static/sentry/app/components/events/errors.tsx (hunk 18 above)
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:157]: i dont know which action is primary in this case unfortunately. I assume Cancel, as it's the safest?

Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:157]: right now buttons are ordered from (most impact) to (least impact)

File: src/sentry/static/sentry/app/components/events/errors.tsx (hunk 19 above)
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:166]: cool will do

**Review by matejminar — APPROVED** [review]

File: src/sentry/api/endpoints/group_reprocessing.py (hunk 5 above)
Comment (jan-auer) [reply src/sentry/api/endpoints/group_reprocessing.py]: Yeah. The idea is that you use `PUT` on resources that you can name and identify, mostly overwriting their (entire) data. For example, writing a file to a location. Some frameworks like Rails also use `PATCH` to do partial updates.

`POST`, on the other hand, is supposed to add to a collection, and you can add the same data multiple times resulting in multiple entries. PUT should not have that behavior.

File: src/sentry/conf/server.py (hunk 7 above)
Comment (jan-auer) [reply src/sentry/conf/server.py]: It's worth noting that this chunk size differs from the one in filestore, which is at 10MB for attachments nowadays. I don't have strong objections against the larger size, and eventually we'll probably streamline. 

Do you see a good place to define this generically?

File: src/sentry/reprocessing2.py (hunk 11 above)
Comment (jan-auer) [reply src/sentry/reprocessing2.py:83]: Agreed. As this remains feature flagged for a while, this gives us a chance to revise this. Let's follow up on this.

File: src/sentry/reprocessing2.py (hunk 12 above)
Comment (jan-auer) [reply src/sentry/reprocessing2.py:101]: This is how blob store works. There is an association between files and blobs, so that you can clone files without re-uploading their blobs. Have a look at [`FileBlobIndex`](https://github.com/getsentry/sentry/blob/bcab979e02fc78c6b24895f94cb7a46762c622cd/src/sentry/models/file.py#L435-L436) if you're interested.

File: src/sentry/conf/server.py (hunk 7 above)
Comment (untitaker) [reply src/sentry/conf/server.py]: place as in which file? probably this one...

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 26]
```diff
@@ -0,0 +1,243 @@
...
+  project: Project;
+  issueId: string;
+};
+
+type State = {
+  isOpen: boolean;
+};
+
+class EventErrors extends React.Component<Props, State> {
+  static propTypes: any = {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/errors.tsx:41]:
```suggestion
  static propTypes = {
```

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 27]
```diff
@@ -0,0 +1,243 @@
...
+    const {api, orgId, project, event} = this.props;
+    const endpoint = `/projects/${orgId}/${project.slug}/events/${event.id}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing event\u2026'));
+
+    try {
+      await api.requestPromise(endpoint, {
+        method: 'POST',
+      });
+    } catch (e) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/errors.tsx:76]:
```suggestion
    } catch {
```

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 28]
```diff
@@ -0,0 +1,243 @@
...
+    const {api, orgId, issueId} = this.props;
+    const endpoint = `/organizations/${orgId}/issues/${issueId}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing issue\u2026'));
+
+    try {
+      await api.requestPromise(endpoint, {
+        method: 'POST',
+      });
+    } catch (e) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/errors.tsx:100]:
```suggestion
    } catch {
```

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 29]
```diff
@@ -0,0 +1,243 @@
...
+            )}
+          </li>
+        </ul>
+      </Body>
+      <Footer>
+        <ButtonBar gap={1}>
+          <Button onClick={this.onReprocessGroup}>
+            {t('Reprocess all events in issue')}
+          </Button>
+          <Button style={{marginLeft: space(1)}} onClick={this.onReprocessEvent}>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/errors.tsx:155]: we usually avoid using `style`. In this case, I would use the `styled` function of emotion js to style this component

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 30]
```diff
@@ -0,0 +1,243 @@
...
+        <BannerSummary>
+          <StyledIconWarning />
+          <span>
+            {tn(
+              'There was %s error encountered while processing this event',
+              'There were %s errors encountered while processing this event',
+              numErrors
+            )}
+          </span>
+          <a data-test-id="event-error-toggle" onClick={this.toggle}>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/errors.tsx:184]: I would use the button component here with the priority link.

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 31]
```diff
@@ -0,0 +1,243 @@
...
+
+  li {
+    margin-bottom: ${space(0.75)};
+    word-break: break-word;
+  }
+
+  pre {
+    background: #f9eded;
+    color: #381618;
+    margin: 5px 0 0;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/errors.tsx:239]:
```suggestion
    margin: ${space(0.5)} 0 0;
```

File: src/sentry/static/sentry/app/components/events/errors.tsx [hunk 32]
```diff
@@ -0,0 +1,243 @@
...
+  padding: ${space(1)} 0 ${space(0.5)} ${space(4)};
+
+  li {
+    margin-bottom: ${space(0.75)};
+    word-break: break-word;
+  }
+
+  pre {
+    background: #f9eded;
+    color: #381618;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/errors.tsx:238]: isn't there any color from the theme that you could use here?

File: src/sentry/conf/server.py (hunk 7 above)
Comment (jan-auer) [reply src/sentry/conf/server.py]: I was thinking about having a config for the generic default cache chunk size. At least, this is a wider configuration than just for reprocessing, as it should be the same with principal ingestion (currently controlled by Relay).

File: src/sentry/static/sentry/app/components/events/errors.tsx (hunk 29 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/components/events/errors.tsx:155]: Those marginLefts can be removed imho now that it's wrapped inside ButtonBar

**Review by tonyo — APPROVED** [review]

File: src/sentry/static/sentry/app/components/events/errors.tsx (hunk 30 above)
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:184]: This code is from the old jsx file, I won't change the UI here

File: src/sentry/static/sentry/app/components/events/errors.tsx (hunk 32 above)
Comment (untitaker) [reply src/sentry/static/sentry/app/components/events/errors.tsx:238]: idk this is from the old code
